torch_geometric/datasets/lrgb.py

```diff
@@ -0,0 +1,278 @@
+
+    @property
+    def processed_file_names(self) -> List[str]:
+        return ['train.pt', 'val.pt', 'test.pt']
+
+    def download(self):
+        shutil.rmtree(self.raw_dir)
+        path = download_url(self.urls[self.name], self.root)
+        extract_zip(path, self.root)
+        os.rename(osp.join(self.root, self.dwnld_file_name[self.name]),
+                  self.raw_dir)
+        os.unlink(path)
+
+    def process(self):
+        if self.name == 'pcqm-contact':
+            # PCQM-Contact
+            self.process_pcqm_contact()
+        else:
+            for split in ['train', 'val', 'test']:
+                if self.name.split('-')[1] == 'sp':
+                    # PascalVOC-SP and COCO-SP
+                    with open(osp.join(self.raw_dir, f'{split}.pickle'),
+                              'rb') as f:
+                        graphs = pickle.load(f)
+                elif self.name.split('-')[0] == 'peptides':
+                    # Peptides-func and Peptides-struct
+                    with open(osp.join(self.raw_dir, f'{split}.pt'),
+                              'rb') as f:
+                        graphs = torch.load(f)
+
+                data_list = []
+                for graph in tqdm(graphs, desc=f'Processing {split} dataset'):
+                    if self.name.split('-')[1] == 'sp':
+                        """
+                        PascalVOC-SP and COCO-SP
+                        Each `graph` is a tuple (x, edge_attr, edge_index, y)
+                            Shape of x : [num_nodes, 14]
+                            Shape of edge_attr : [num_edges, 2]
+                            Shape of edge_index : [2, num_edges]
+                            Shape of y : [num_nodes]
+                        """
+                        x = graph[0].to(torch.float)
+                        edge_attr = graph[1].to(torch.float)
+                        edge_index = graph[2]
+                        y = torch.LongTensor(graph[3])
+                    elif self.name.split('-')[0] == 'peptides':
+                        """
+                        Peptides-func and Peptides-struct
+                        Each `graph` is a tuple (x, edge_attr, edge_index, y)
+                            Shape of x : [num_nodes, 9]
+                            Shape of edge_attr : [num_edges, 3]
+                            Shape of edge_index : [2, num_edges]
+                            Shape of y : [1, 10] for Peptides-func,  or
+                                         [1, 11] for Peptides-struct
+                        """
+                        x = graph[0]
+                        edge_attr = graph[1]
+                        edge_index = graph[2]
+                        y = graph[3]
+
+                    if self.name == 'coco-sp':
+                        # Label remapping for coco-sp.
+                        # See self.label_remap_coco() func
+                        label_map = self.label_remap_coco()
```

Comment:
We are not necessarily to call this method and get `label_map` at each iteration, right?

Reply:
Right -thanks! It can be called just once for coco. I updated the file.